Assets/Scripts/GameController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class GameController : MonoBehaviour
+{
+
+    public static GameController Instance { get; private set; }
+
+    public bool IsPaused { get; private set; }
+
+    public bool IsGameOver { get; private set; }
+
+    // Maps the level with the upcoming one
+    private Dictionary<string, string> nextLevelsDictionary = new Dictionary<string, string>()
+    {
+        { "Level1", "Level2" }
+    };
+
+    void Awake()
+    {
+        if (Instance == null)
+        {
+            Instance = this;
+        }
+        else if (Instance != this)
+        {
+            Destroy(this);
+        }
+        DontDestroyOnLoad(gameObject);
+    }
+
+    public void Pause(bool shouldPause)
+    {
+        IsPaused = shouldPause;
+        if (shouldPause)
+        {
+            Time.timeScale = 0.0f;
+            // TODO bring up Pause menu
+        }
+        else
+        {
+            Time.timeScale = 1.0f;
+            // TODO shut down Pause menu
+        }
+    }
```

Comment:
Pause method is fine, but how about making it like a toggle function and passing nothing in parameters?

if(IsPaused)
//pause
else
//unpause

then we could more easily use this method in Update() or other methods